packages/dds/tree/src/test/simple-tree/arrayNode.spec.ts

```diff
@@ -0,0 +1,62 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { strict as assert } from "assert";
+
+import {
+	ImplicitAllowedTypes,
+	InsertableTreeFieldFromImplicitField,
+	InsertableTreeNodeFromImplicitAllowedTypes,
+	NodeKind,
+	SchemaFactory,
+	TreeArrayNode,
+	TreeFieldFromImplicitField,
+	TreeNodeSchema,
+} from "../../simple-tree/index.js";
+
+import { hydrate } from "./utils.js";
+import { Mutable } from "../../util/index.js";
+
+type ArrayNodeSchema<T extends ImplicitAllowedTypes> = TreeNodeSchema<
+	string,
+	NodeKind.Array,
+	TreeArrayNode<T>,
+	Iterable<InsertableTreeNodeFromImplicitAllowedTypes<T>>,
+	true
+>;
+
+const schemaFactory = new SchemaFactory("Test");
+
+function hydrateArray<T extends TreeNodeSchema | readonly TreeNodeSchema[]>(
+	schemaType: "structural" | "class-based",
+	contentType: T,
+	content?: () => InsertableTreeFieldFromImplicitField<ArrayNodeSchema<T>>,
+): TreeFieldFromImplicitField<ArrayNodeSchema<T>> {
+	const schema =
+		schemaType === "structural"
+			? schemaFactory.array(contentType)
+			: schemaFactory.array("Array", contentType);
+
+	return hydrate(schema, content?.() ?? []);
+}
+
+describe("ArrayNode", () => {
+	function describeArrayNodeTests(
```

Comment:
I find this pattern hard to follow, and it breaks things like .only, and .skip.

I think these tests would be much easier to read if they just declared the schema normally and used them.

If you really want to look over two different array schema, you can just declare both of them, then loop over them:

class SchemaA ....
const SchemaB = ...

for (schema of [SchemA, SchemaB]) {...}

That keeps the code and typing much closer to how real usage will work, and is much easier to follow and avoids things the interfere with test tooling.

Reply:
I think you are conflating this pattern with the "make it()-style wrappers" anti-pattern that we've previously removed from our codebase. This is not quite the same thing. 

In the antipattern, we generate `it()` tests from a common helpers (a code smell for detecting this is that the "title" of each `it()` is generated dynamically). So it breaks linking from the test browser, it breaks only and it breaks skip, because there is only a single actual call to `it()` in the source code, which runs many times.

What I've done here is create two _describe_ blocks, which are _not_ generated dynamically, and I run the same list of `it()`s (currently just one) twice, one in each describe. It is essentially just running the tests in two different modes. So linking to the test browser still brings you to the correct test, and skip/only will still only skip/only the correct test. I will concede that if you skip a test, it will skip both "modes" rather than you being able to skip just one mode. But the upside is that we don't have to wrap every test in a loop, and the two cases are automatically split into different describe blocks.

![image](https://github.com/microsoft/FluidFramework/assets/78610362/e3085b9a-0d42-486e-88b6-3da7ec6e8c8b)

Reply:
I've made the layout even simpler by pulling out the describe blocks. Now there is a logical place to put 
1. Tests that are only relevant to "structural-schema" ArrayNodes
2. Tests that are only relevant to "class-based-schema" ArrayNodes
3. Tests that are relevant to both

It should be simple to follow the test structure now.